components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/dao/impl/SAML2ArtifactInfoDAOImpl.java

```diff
@@ -0,0 +1,192 @@
+                    {
+                        try {
+                            return new SAML2ArtifactInfo(resultSet.getInt(1),
+                                    (SAMLSSOAuthnReqDTO) getBlobObject(resultSet.getBinaryStream(2)),
+                                    resultSet.getString(3),
+                                    new DateTime(resultSet.getTimestamp(4)),
+                                    new DateTime(resultSet.getTimestamp(5)));
+                        } catch (IOException | ClassNotFoundException e) {
+                            throw new SQLException(e);
+                        }
+                    },
+                    preparedStatement -> {
+                        preparedStatement.setBytes(1, sourceId);
+                        preparedStatement.setBytes(2, messageHandler);
+
+                    });
+        } catch (DataAccessException e) {
+            throw new ArtifactBindingException("Error while retrieving SAML2 artifact information.", e);
+        }
+
+        // Deleting artifact record.
+        if (saml2ArtifactInfo != null) {
+
+            try {
+                jdbcTemplate.executeUpdate(ASSERTION_DELETE_SQL, preparedStatement ->
+                        preparedStatement.setInt(1, saml2ArtifactInfo.getId()));
+            } catch (DataAccessException e) {
+                throw new ArtifactBindingException("Error while retrieving SAML2 artifact information.", e);
+            }
+        }
+
+        return saml2ArtifactInfo;
+    }
+
+    @Override
+    public Assertion getSAMLAssertion(String assertionId) throws ArtifactBindingException {
+
+        final String ASSERTION_RETRIVE_QUERY =
+                "SELECT SAML2_ASSERTION FROM IDN_SAML2_ASSERTION_STORE WHERE SAML2_ID=?";
+
+        JdbcTemplate jdbcTemplate = JdbcUtils.getNewTemplate();
+        Assertion assertion;
+
+        try {
+            String assertionString = jdbcTemplate.fetchSingleRecord(ASSERTION_RETRIVE_QUERY, (resultSet, rowNumber) ->
+                            resultSet.getString(1), preparedStatement ->
+                    preparedStatement.setString(1, assertionId));
+            assertion = (Assertion) SAMLSSOUtil.unmarshall(assertionString);
+        } catch (DataAccessException e) {
+            throw new ArtifactBindingException("Error while retrieving SAML2 artifact information.", e);
+        } catch (IdentityException e) {
+            throw new ArtifactBindingException("Error while unmarshalling SAML assertion.", e);
+        }
+
+        return assertion;
+    }
+
+    /**
+     * Serialize an object and set into a prepared statement as a blob.
+     *
+     * @param prepStmt Prepared statement.
+     * @param value    Object to be saved.
+     * @param index    Index of the prepared statement.
+     * @throws SQLException
+     * @throws IOException
+     */
+    private void setBlobObject(PreparedStatement prepStmt, Object value, int index) throws SQLException, IOException {
```

Comment:
Please create an issue so that this code can be moved to JDBCTemplate later.

Reply:
Added. https://github.com/wso2/product-is/issues/3441